packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -3211,19 +3170,17 @@ function commitRootImpl(
     if (enableSchedulingProfiler) {
       markCommitStopped();
     }
-    return null;
+    return;
   } else {
     if (__DEV__) {
       if (lanes === NoLanes) {
         console.error(
-          'root.finishedLanes should not be empty during a commit. This is a ' +
+          'finishedLanes should not be empty during a commit. This is a ' +
             'bug in React.',
         );
       }
     }
   }
-  root.finishedWork = null;
-  root.finishedLanes = NoLanes;
```

Comment:
Should `root.cancelPendingCommit` and `timeoutHandle` get reset?

Reply:
`root.cancelPendingCommit` gets reset in the beginning of `commitRoot`. `timeoutHandle` gets reset in the beginning of `commitRootWhenReady`.

Which is really just right at the beginning of the callback that was pending.

The idea is that it's reset right after the callback is fired, at the very beginning, since it's no longer pending. Or it gets reset when it's cancelled.